tests/analytical.py

```diff
@@ -1,88 +1,113 @@
 import numpy as np
 
 
-def gradient(x_a, y_a, z_a, x_b, y_b, z_b, c_a, c_b):
+def gradient(coordinates,charges):
     """Computes the gradient."""
+    number_coordinates = len(coordinates)
+    eval_E = []
```

Comment:
We know already the size of the gradient vector: 3 times the number of atoms (the x, y, z derivatives for each atom) We can use a numpy array directly, rather than first a list and then a `np.array`:
```python
eval_E = np.zeros_like(coordinates)
```
`zeros_like` creates a numpy array of zeros with the same shape (number of rows and columns) of the parameter. In this case, the gradient will be a matrix like the coordinates matrix.
Documentation for `zeros_like` here: https://numpy.org/doc/stable/reference/generated/numpy.zeros_like.html